compiler/AST/FnSymbol.cpp

```diff
@@ -786,11 +820,18 @@ bool FnSymbol::hasGenericFormals(SymbolMap* map) const {
         USR_PRINT(this,
                   "   formal argument '%s' causes it to be", formal->name);
       }
-      return true; // no need to examine the remaining formals
     }
+    anyGeneric = anyGeneric || isGeneric;
+    if (isGeneric) {
+      while ((int)formalIsGeneric.size() < count) {
+        formalIsGeneric.push_back(0);
+      }
+      formalIsGeneric.push_back(isGeneric?1:0);
```

Comment:
You can just push_back a `1` in the above.
OK to leave as-is, please double-check correctness.